crates/matrix-sdk-crypto/src/olm/session.rs

```diff
@@ -237,23 +233,17 @@ impl Session {
     ///
     /// * `pickle_mode` - The mode that was used to pickle the session, either
     /// an unencrypted mode or an encrypted using passphrase.
-    pub fn from_pickle(
-        user_id: OwnedUserId,
-        device_id: OwnedDeviceId,
-        our_identity_keys: Arc<IdentityKeys>,
-        pickle: PickledSession,
-    ) -> Self {
+    pub fn from_pickle(device_keys: DeviceKeys, pickle: PickledSession) -> Self {
+        // FIXME: assert that device_keys has curve25519 and ed25519 keys
```

Comment:
Is this a FIXME we should do before merging?

Reply:
Right, I meant to put a comment about that in the PR.  I'm wondering if we should change the return value to a `Result`, or if we should just do an `unwrap` or `expect` and panic if the keys aren't there.  We should be pretty safe for the time being, but I'm a bit worried that if we change crypto systems in the future, this will come back and bite us.  I would lean towards just doing an `unwrap`/`expect`, but I wanted a second opinion.

Reply:
When in doubt, prefer `Result`, I'd say.